src/components/Label.js

```diff
@@ -0,0 +1,53 @@
+/*
+ * OS.js - JavaScript Cloud/Web Desktop Platform
+ *
+ * Copyright (c) 2011-2019, Anders Evenrud <andersevenrud@gmail.com>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @author  Anders Evenrud <andersevenrud@gmail.com>
+ * @licence Simplified BSD License
+ */
+
+import {h} from 'hyperapp';
+import {Element} from './Element';
+
+/**
+ * A flexbox
+ * @param {Object} props Properties
+ * @param {string} [props.text] Label Text
+ * @param {string} [props.placement] Placement
+ * @param {h[]} children Children
+ */
+export const Label = (props = {}, children = []) => {
```

Comment:
After looking at this now a second time, I suggest changing it to this so it is consistent with all the other components.

* Moved Element properties into `box` (aka BoxProperties)
* Added `for` property support

Noticed that the description of the comment was wrong, so updated that as well :)

```javascript
/**
 * A label element
 * @param {Object} props Properties
 * @param {string} [props.text] Label Text
 * @param {string} [props.placement] Placement
 * @param {string} [props.for] The "for" attribute
 * @param {BoxProperties} [props.box] Box Properties
 * @param {h[]} children Children
 */
export const Label = (props = {}, children = []) => {
  const placement = props.placement || 'top';
  const text = props.text || '';

  const elementProps = Object.assign({
    class: ['osjs-gui-field-label', 'osjs-gui-field-label-on-' + placement]
  }, props.box || {});

  return h(Element, elementProps, [
    h('label', {for: props.for}, text),
    children
  ]);
};
```

Reply:
Yes, another casualty of copy/paste 🐑

I will add these changes, thank you. So what is the semantic meaning of `BoxProperties`? Is that just a name for the properties that a standard `Box` would have?

Reply:
Indeed. It's just the shape of a specific set of properties used for any `Box` derivative. Any "container" element will have these properties directly on the component, while others have it as a sub-property named `box`.

Full definition here: https://github.com/os-js/osjs-gui/blob/master/src/components/Element.js#L33

Reply:
Speaking of copy/paste --- `Window dimension definition` :man_facepalming: 

That's straight from `@osjs/client` `window.js` source file :laughing:

Reply:
Haha, I guess it happens to the greats as well. I will make these changes and test - thanks for the explanation.